version: '3.8'

services:
  web:
    build:
      context: ./auth
      dockerfile: Dockerfile
    command: bash -c "python manage.py makemigrations && python manage.py migrate && python manage.py runserver 0.0.0.0:8000"
    volumes:
      - ./auth:/app
    ports:
      - "8000:8000"
    environment:
      - DJANGO_SETTINGS_MODULE=auth.settings
    depends_on:
      - otp_service

  test:
    build:
      context: ./auth
      dockerfile: Dockerfile
    command: bash -c "pytest && python manage.py behave"
    environment:
      - DJANGO_SETTINGS_MODULE=auth.settings
    depends_on:
      - otp_service

  otp_service:
    build:
      context: ./notification
      dockerfile: Dockerfile
    command: python otp.py
    volumes:
      - ./notification:/app
    ports:
      - "50051:50051"
    environment:
      - CONFIG_FILE=/app/config.ini
    healthcheck:
      test: [ "CMD", "grpc_health_probe", "-addr=localhost:50051" ]
      interval: 10s
      timeout: 5s
      retries: 3